hibernate-core/src/main/java/org/hibernate/type/descriptor/sql/SqlTypeDescriptor.java

```diff
@@ -95,6 +95,48 @@ default int getJdbcTypeCode() {
 	 */
 	<X> ValueExtractor<X> getExtractor(JavaTypeDescriptor<X> javaTypeDescriptor);
 
+	default boolean isInteger() {
+		switch ( getSqlType() ) {
+			case Types.BIT:
+			case Types.TINYINT:
+			case Types.SMALLINT:
+			case Types.INTEGER:
+			case Types.BIGINT:
+				return true;
+		}
+		return false;
+	}
+
+	default boolean isNumber() {
+		switch ( getSqlType() ) {
+			case Types.BIT:
+			case Types.TINYINT:
+			case Types.SMALLINT:
+			case Types.INTEGER:
+			case Types.BIGINT:
+			case Types.FLOAT:
+			case Types.REAL:
+			case Types.DOUBLE:
+			case Types.DECIMAL:
+			case Types.NUMERIC:
+				return true;
+		}
+		return false;
+	}
+
+	default boolean isString() {
+		switch ( getSqlType() ) {
+			case Types.CHAR:
+			case Types.NCHAR:
+			case Types.VARCHAR:
+			case Types.NVARCHAR:
+			case Types.LONGVARCHAR:
+			case Types.LONGNVARCHAR:
+				return true;
+		}
+		return false;
+	}
+
```

Comment:
In one of my PRs I have added functions like these to `Dialect`, so we need to unify them in some `Helper` class I guess.

Reply:
I suppose you mean the changes from https://github.com/hibernate/hibernate-orm/pull/3758 ? Yeah we need to unify that somehow. I think we can use these methods I introduced in this PR though, if we move the `getCheckCondition` method to `BasicType` like I suggested.